pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -113,6 +124,65 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
+	if accountClaim.Spec.BYOC {
+
+		if accountClaim.Spec.AccountLink == "" {
+
+			//Create a new account with BYOC flag
+			newAccount := utils.NewAccountForCR(awsv1alpha1.AccountCrNamespace)
+			newAccount.Spec.BYOC = true
+			newAccount.Spec.AwsAccountID = accountClaim.Spec.BYOCAWSAccountID
+			newAccount.Spec.ClaimLink = accountClaim.ObjectMeta.Name
+			newAccount.Spec.ClaimLinkNamespace = accountClaim.ObjectMeta.Namespace
+			utils.AddFinalizer(newAccount, "finalizer.aws.managed.openshift.io")
+
+			// Set AccountClaim instance as the owner and controller
+			if err := controllerutil.SetControllerReference(accountClaim, newAccount, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Create the new account
+			err = r.client.Create(context.TODO(), newAccount)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Set the accountLink of the AccountClaim to the new account if create is successful
+			accountClaim.Spec.AccountLink = newAccount.Name
+			err = r.client.Update(context.TODO(), accountClaim)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, nil
+		}
+
+		// Get the account and check if its Ready
+		byocAccount := &awsv1alpha1.Account{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: accountClaim.Spec.AccountLink, Namespace: awsv1alpha1.AccountCrNamespace}, byocAccount)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		reqLogger.Info(fmt.Sprintf("Byoc account state: %+v", byocAccount.Status.State))
+		if byocAccount.Status.State == string(awsv1alpha1.AccountReady) {
+
+			reqLogger.Info("Ready")
```

Comment:
and this

Reply:
Good point @sam-nguyen7, @fahlmant this could be more descriptive or removed?

Reply:
Needs to be removed